### PR TITLE
Fix #44 to support XA Connection Factory

### LINKED ISSUE #44: Support XAConnectionFactory
I'd like introduce an option `quarkus.artemis.xa=[true|false]` and if it is `true`, `ArtemisJmsProcessor` will produce a `XAConnectionFatory` bean.

```
public Supplier<XAConnectionFactory> getXAConnectionFactorySupplier() {
        return new Supplier<XAConnectionFactory>() {
            @Override
            public XAConnectionFactory get() {
                return new ActiveMQXAConnectionFactory(config.url, config.username.orElse(null), config.password.orElse(null));
            }
        };
    }
```

### PR DESCRIPTION
**Pull request by zhfeng**

### CODE REVIEW

File: integration-tests/jms/pom.xml [hunk 1]
```diff
@@ -18,6 +18,12 @@
             <artifactId>quarkus-resteasy</artifactId>
         </dependency>
 
+        <!-- Narayana Transaction Manager -->
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-narayana-jta</artifactId>
+        </dependency>
```
Comment (middagj) [inline integration-tests/jms/pom.xml:25]: I think this should be optional

File: integration-tests/jms/src/main/java/io/quarkus/it/artemis/ArtemisProducerManager.java [hunk 2]
```diff
@@ -5,17 +5,51 @@
...
     public void send(String body) {
         try (JMSContext context = connectionFactory.createContext(JMSContext.AUTO_ACKNOWLEDGE)) {
-            JMSProducer producer = context.createProducer();
-            producer.send(context.createQueue("test-jms"), body);
+            _send(context, body);
+        }
+    }
+
+    public void sendXA(String body) throws Exception {
+        XAJMSContext context = xaConnectionFactory.createXAContext();
```
Comment (middagj) [inline integration-tests/jms/src/main/java/io/quarkus/it/artemis/ArtemisProducerManager.java:37]: shouldn't this also be in a try with resources?

**Review by middagj — COMMENTED** [review]
Some small things, but looks good to me.

File: integration-tests/jms/pom.xml (hunk 1 above)
Comment (zhfeng) [reply integration-tests/jms/pom.xml:25]: Good catch!

File: integration-tests/jms/src/main/java/io/quarkus/it/artemis/ArtemisProducerManager.java (hunk 2 above)
Comment (zhfeng) [reply integration-tests/jms/src/main/java/io/quarkus/it/artemis/ArtemisProducerManager.java:37]: Anyway, this should be better processed by other wrapper, such as `org.messaginghub:pooled-jms`.

**Comment by zhfeng** [comment]
@middagj I also need it for working on https://github.com/quarkusio/quarkus/issues/14871

File: integration-tests/jms/src/main/java/io/quarkus/it/artemis/ArtemisProducerManager.java (hunk 2 above)
Comment (middagj) [reply integration-tests/jms/src/main/java/io/quarkus/it/artemis/ArtemisProducerManager.java:37]: Okay, not an expert on XA and it is test code, so should be fine.

**Pull request merged**